libs/core/langchain_core/tracers/event_stream.py

```diff
@@ -116,41 +118,83 @@ async def tap_output_aiter(
         self, run_id: UUID, output: AsyncIterator[T]
     ) -> AsyncIterator[T]:
         """Tap the output aiter."""
-        async for chunk in output:
-            run_info = self.run_map.get(run_id)
-            if run_info is None:
-                raise AssertionError(f"Run ID {run_id} not found in run map.")
-            self._send(
-                {
-                    "event": f"on_{run_info['run_type']}_stream",
-                    "data": {"chunk": chunk},
-                    "run_id": str(run_id),
-                    "name": run_info["name"],
-                    "tags": run_info["tags"],
-                    "metadata": run_info["metadata"],
-                },
-                run_info["run_type"],
-            )
-            yield chunk
+        sentinel = object()
+        # atomic check and set
+        tap = self.is_tapped.setdefault(run_id, sentinel)
+        # wait for first chunk
+        first = await py_anext(output, sentinel)
+        if first is sentinel:
+            return
+        # get run info
+        run_info = self.run_map.get(run_id)
+        if run_info is None:
+            # run has finished, don't issue any stream events
+            yield first
```

Comment:
no need to raise exception here, if something not working we already raise exception in the end events